controllers/chat/moveToAnon.js

```diff
@@ -196,38 +211,23 @@ const moveToAnon = async (req, res) => {
       console.log(e);
     }
 
-    const targetsUserModel = await UsersFunction.findMultipleUsersById(User, members);
-    targetsUserModel.push({
-      user_id: req.userId,
-      username: `Anonymous ${anon_user_info_emoji_name}`,
-      profile_pic_path: '',
-      anon_user_info_color_code,
-      anon_user_info_color_name,
-      anon_user_info_emoji_code,
-      anon_user_info_emoji_name
-    });
-
     // get 100 messages
     const channelFilters = {cid: 'messaging:' + newChannel.id};
     const messageFilters = {created_at: {$lte: new Date()}};
     const messageHistory = await client.search(channelFilters, messageFilters, {
       sort: [{updated_at: -1}],
       limit: 100
     });
-
     await client.disconnectUser();
-
     const response = {
       ...createdChannel,
       better_channel_members: Object.values(newStateMemberWithAnonInfo),
-      better_channel_members_objet: newStateMemberWithAnonInfo,
+      better_channel_members_object: newStateMemberWithAnonInfo,
       messageHistories: messageHistory.results
     };
-
     if (createdChannel?.channel?.is_channel_blocked) {
       return res.status(400).json(responseError('Channel is blocked', response));
     }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In this section, you're directly accessing properties on objects without checking if the object exists first. This can lead to `TypeError: Cannot read property 'property' of undefined` errors. It's a good practice to check if an object exists before trying to access its properties.

For example, in line 223, you're accessing `createdChannel`. If `createdChannel` is null or undefined, this will throw an error. The same applies for line 229.

Consider adding checks for these objects before accessing their properties.

```diff
- ...createdChannel,
+ ...createdChannel || {},
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->